pkg/services/pluginsintegration/clientmiddleware/utils.go

```diff
@@ -23,3 +39,36 @@ type callResourceResponseSenderFunc func(res *backend.CallResourceResponse) erro
 func (fn callResourceResponseSenderFunc) Send(res *backend.CallResourceResponse) error {
 	return fn(res)
 }
+
+func requestStatusFromError(err error) (requestStatus, error) {
+	status := requestStatusOK
+	if err != nil {
+		status = requestStatusError
+		if errors.Is(err, context.Canceled) {
+			status = requestStatusCancelled
+		}
+	}
+
+	return status, err
+}
+
+func requestStatusFromQueryDataResponse(res *backend.QueryDataResponse, err error) (requestStatus, error) {
+	if err != nil {
+		return requestStatusFromError(err)
+	}
+
+	status := requestStatusOK
+
+	if res != nil {
+		for _, dr := range res.Responses {
+			if dr.Error != nil {
+				s, _ := requestStatusFromError(dr.Error)
+				if s > status {
+					status = s
+				}
```

Comment:
does it make sense to keep iterating if one error is found?

Reply:
Good question. In general we should probably stop, but then we need yet another check. I'll consider it

Reply:
Fixed

Reply:
Made error highest priority instead of cancelled. Should make more sense I think